apps/aehyperchains/src/aehc_consensus_hyperchains.erl

```diff
@@ -1039,9 +1062,77 @@ protocol_staking_contract_call(Trees0, TxEnv, Query) ->
         {error, _What} = Err -> Err
     end.
 
+
+%% Implementation
+
+-spec load_hc_staking_contract() -> ok | no_return().
+load_hc_staking_contract() ->
+    P = aec_fork_block_settings:hc_staking_contract_file(),
+    case filelib:is_file(P) of
+        true -> ok;
+        false ->
+            aec_consensus:config_assertion_failed(
+                "Hyperchains staking contract not present.",
+                " Terminating the node - file not found ~p", [P])
+    end,
+    lager:debug("Loading hyperchains staking contract"),
+    {ok, Data} = file:read_file(P),
+    try
+        #{
+            <<"bytecode">> := B,
+            <<"aci">> := JText
+        } = jsx_decode(Data),
+        CompiledACI = aeaci_aci:from_string(JText, #{backend => fate}),
+        {contract_bytearray, Bytecode} = decode(B),
+        persistent_term:put(?STAKING_CONTRACT, #{bytecode => Bytecode, aci => CompiledACI})
+    catch E:R:Stacktrace ->
+        aec_consensus:config_assertion_failed(
+            "Hyperchains staking contract is malformed.",
+            " Terminating the node - ~p ~p ~p", [E, R, Stacktrace])
+    end.
+
 %% What consensus shall we use in case the activation criteria of the hyperchain consensus weren't meet?
 %% TODO: customize
-fallback_consensus() ->
-    aec_consensus_bitcoin_ng.
+fallback_consensus() -> aec_consensus_bitcoin_ng.
 
+-spec hc_genesis_version() -> non_neg_integer().
 hc_genesis_version() -> ?HC_GENESIS_VERSION.
+
+
+%% Helpers
+
+-spec apply_fallback(fallback_funs()) -> term().
+apply_fallback(Function) -> apply_fallback(Function, []).
+
+-spec apply_fallback(fallback_funs(), [term()]) -> term().
+apply_fallback(Function, Args) -> apply(fallback_consensus(), Function, Args).
+
+-spec jsx_decode(jsx:json_text()) -> map() | no_return().
+jsx_decode(Data) ->
+    case jsx:decode(Data, [{return_maps, true}, {labels, binary}]) of
+        Map when is_map(Map) -> Map;
+        [Map] when is_map(Map) -> Map
+    end.
+
+-spec encode_contract_pubkey(pubkey()) -> binary().
+encode_contract_pubkey(Pubkey) -> aeser_api_encoder:encode(contract_pubkey, Pubkey).
+
+-spec encode_account_pubkey(pubkey()) -> binary().
+encode_account_pubkey(Pubkey) -> aeser_api_encoder:encode(account_pubkey, Pubkey).
+
+%% Typing here is not perfect, we need to have known_type() and payload()
+%% exported from aeser_api_encoder
+-spec decode(binary()) -> {atom(), binary()}.
```

Comment:
This is same case as `ldebug` helper. Remove this helper as it decreases readability.